test/test_any_automated_install.py

```diff
@@ -548,24 +536,17 @@ def test_installPihole_fresh_install_readableBlockpage(host, test_webpage):
     # change nameserver to pi-hole
     # setting nameserver in /etc/resolv.conf to pi-hole does
     # not work here because of the way docker uses this file
-    ns = host.run(r"sed -i 's/nameserver.*/nameserver 127.0.0.1/' /etc/resolv.conf")
+    ns = host.run(
```

Comment:
>On Fedora-based systems, not sure why sed rename is failing in the tests. I'll continue debugging later.

I just did an interactive rebase to remove this commit locally, and then ran tests with:

```text
adam@adam-pc:~/pi-hole$ tox -c test/tox.fedora_35.ini
[snip]
============================================================================== 43 passed in 86.62s (0:01:26) ===============================================================================
adam@adam-pc:~/pi-hole$ tox -c test/tox.fedora_36.ini
[snip]
============================================================================== 43 passed in 82.76s (0:01:22) ===============================================================================
adam@adam-pc:~/pi-hole$ tox -c test/tox.fedora_37.ini
[snip]
============================================================================== 43 passed in 89.45s (0:01:29) ===============================================================================
```

Whatever you saw in the test runs earlier, I guess it was transient, as I'm unable to repro. Please revert this change

Reply:
See https://github.com/pi-hole/pi-hole/pull/5075#issuecomment-1357307894
`stderr=b'sed: cannot rename /etc/sedtvsuT3: Device or resource busy\n'`

Please double-check your logs.  View the raw logs and search for: "sed: cannot rename"
I think that you will see the error in your logs, even if things happen to continue to work.

Please confirm if you want me to remove the commit.  Thanks.

Reply:
That's the comment I was basing my curiosity on. Aside from the error you saw, I could not see a reason to change this part of the tests, and so investigated myself locally.

Fortunately, I still have the terminal open - No such error :) (please revert the change)

```text
adam@adam-pc:~/pi-hole$ git rebase -i 0d5d3a1b2209bacbbdb09c069ce81cf2f3dafb28
warning: skipped previously applied commit 440e49b1
hint: use --reapply-cherry-picks to include skipped commits
hint: Disable this message with "git config advice.skippedCherryPicks false"
Successfully rebased and updated refs/heads/pr/gstrauss/5075.
adam@adam-pc:~/pi-hole$ tox -c test/tox.fedora_35.ini 
py3: install_deps /home/adam/pi-hole/test> python -I -m pip install -r requirements.txt
.pkg: install_requires /home/adam/pi-hole/test> python -I -m pip install 'setuptools>=40.8.0' wheel
.pkg: _optional_hooks /home/adam/pi-hole/test> python /home/adam/.local/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: get_requires_for_build_sdist /home/adam/pi-hole/test> python /home/adam/.local/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: install_requires_for_build_sdist /home/adam/pi-hole/test> python -I -m pip install pytest-runner
.pkg: prepare_metadata_for_build_wheel /home/adam/pi-hole/test> python /home/adam/.local/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: build_sdist /home/adam/pi-hole/test> python /home/adam/.local/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
py3: install_package /home/adam/pi-hole/test> python -I -m pip install --force-reinstall --no-deps /home/adam/pi-hole/test/.tox/.tmp/package/1/UNKNOWN-0.0.0.tar.gz
py3: commands[0] /home/adam/pi-hole/test> docker build -f _fedora_35.Dockerfile -t pytest_pihole:test_container ../
[+] Building 4.6s (12/12) FINISHED 

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
py3: commands[1] /home/adam/pi-hole/test> pytest -vv -n auto ./test_any_automated_install.py ./test_any_utils.py ./test_centos_fedora_common_support.py ./test_fedora_support.py
=================================================================================== test session starts ====================================================================================
platform linux -- Python 3.10.6, pytest-7.2.0, pluggy-1.0.0 -- /home/adam/pi-hole/test/.tox/py3/bin/python
cachedir: .tox/py3/.pytest_cache
rootdir: /home/adam/pi-hole/test
plugins: testinfra-7.0.0, xdist-3.1.0
[gw0] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw1] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw2] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw3] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw4] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw5] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw6] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw7] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw8] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw9] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw10] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw11] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw12] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw13] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw14] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw15] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw0] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                     
[gw1] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                      
[gw2] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                       
[gw3] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                            
[gw4] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                               
[gw5] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                    
[gw6] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                       
[gw7] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                          
[gw8] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                             
[gw9] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                
[gw10] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                
[gw11] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                     
[gw12] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                        
[gw13] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                           
[gw14] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                              
[gw15] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                                 
gw0 [43] / gw1 [43] / gw2 [43] / gw3 [43] / gw4 [43] / gw5 [43] / gw6 [43] / gw7 [43] / gw8 [43] / gw9 [43] / gw10 [43] / gw11 [43] / gw12 [43] / gw13 [43] / gw14 [43] / gw15 [43]
scheduling tests via LoadScheduling

test_any_automated_install.py::test_installPiholeWeb_fresh_install_no_errors 
test_any_automated_install.py::test_FTL_detect_armv6l_no_errors 
test_any_automated_install.py::test_IPv6_only_ULA 
test_any_automated_install.py::test_FTL_detect_armv8a_no_errors 
test_any_automated_install.py::test_supported_package_manager 
test_any_automated_install.py::test_FTL_detect_unknown_no_errors 
test_any_automated_install.py::test_setupVars_saved_to_file 
test_any_automated_install.py::test_validate_ip 
test_any_automated_install.py::test_installPihole_fresh_install_readableBlockpage[True] 
test_any_automated_install.py::test_FTL_detect_armv4t_no_errors 
test_any_automated_install.py::test_FTL_binary_installed_and_responsive_no_errors 
test_any_automated_install.py::test_update_package_cache_failure_no_errors 
test_any_automated_install.py::test_os_check_passes 
test_any_automated_install.py::test_package_manager_has_pihole_deps 
test_any_utils.py::test_key_val_replacement_works 
test_any_automated_install.py::test_IPv6_GUA_ULA_test 
[gw2] [  2%] PASSED test_any_automated_install.py::test_installPiholeWeb_fresh_install_no_errors 
[gw10] [  4%] PASSED test_any_automated_install.py::test_IPv6_only_ULA 
test_any_automated_install.py::test_installPihole_fresh_install_readableFiles 
[gw4] [  6%] PASSED test_any_automated_install.py::test_update_package_cache_failure_no_errors 
[gw11] [  9%] PASSED test_any_automated_install.py::test_IPv6_GUA_ULA_test 
[gw0] [ 11%] PASSED test_any_automated_install.py::test_supported_package_manager 
test_any_automated_install.py::test_IPv6_only_GUA 
test_any_automated_install.py::test_FTL_detect_aarch64_no_errors 
[gw1] [ 13%] PASSED test_any_automated_install.py::test_setupVars_saved_to_file 
[gw15] [ 16%] PASSED test_any_utils.py::test_key_val_replacement_works 
test_any_automated_install.py::test_IPv6_ULA_GUA_test 
test_any_automated_install.py::test_setupVars_are_sourced_to_global_scope 
test_any_automated_install.py::test_selinux_not_detected 
test_any_utils.py::test_key_addition_works 
[gw10] [ 18%] PASSED test_any_automated_install.py::test_IPv6_only_GUA 
[gw1] [ 20%] PASSED test_any_automated_install.py::test_selinux_not_detected 
[gw15] [ 23%] PASSED test_any_utils.py::test_key_addition_works 
test_any_utils.py::test_getFTLAPIPort_default 
[gw0] [ 25%] PASSED test_any_automated_install.py::test_setupVars_are_sourced_to_global_scope 
[gw11] [ 27%] PASSED test_any_automated_install.py::test_IPv6_ULA_GUA_test 
test_any_utils.py::test_getFTLPID_default 
test_any_utils.py::test_getFTLPIDFile_and_getFTLPID_custom 
test_any_utils.py::test_getFTLPIDFile_default 
test_any_utils.py::test_getFTLAPIPort_malicious 
[gw9] [ 30%] PASSED test_any_automated_install.py::test_FTL_binary_installed_and_responsive_no_errors 
[gw10] [ 32%] PASSED test_any_utils.py::test_getFTLAPIPort_default 
[gw1] [ 34%] PASSED test_any_utils.py::test_getFTLPID_default 
[gw5] [ 37%] PASSED test_any_automated_install.py::test_FTL_detect_armv4t_no_errors 
[gw15] [ 39%] PASSED test_any_utils.py::test_getFTLPIDFile_and_getFTLPID_custom 
[gw12] [ 41%] PASSED test_any_automated_install.py::test_validate_ip 
test_any_automated_install.py::test_IPv6_only_link_local 
[gw0] [ 44%] PASSED test_any_utils.py::test_getFTLPIDFile_default 
test_centos_fedora_common_support.py::test_selinux_permissive 
[gw8] [ 46%] PASSED test_any_automated_install.py::test_FTL_detect_unknown_no_errors 
test_centos_fedora_common_support.py::test_selinux_enforcing_exit 
test_any_automated_install.py::test_FTL_detect_armv5te_no_errors 
test_centos_fedora_common_support.py::test_selinux_disabled 
test_any_automated_install.py::test_os_check_fails 
[gw14] [ 48%] PASSED test_any_automated_install.py::test_package_manager_has_pihole_deps 
test_fedora_support.py::test_epel_and_remi_not_installed_fedora 
[gw11] [ 51%] PASSED test_any_utils.py::test_getFTLAPIPort_malicious 
test_any_automated_install.py::test_FTL_download_aarch64_no_errors 
test_any_automated_install.py::test_package_manager_has_web_deps 
[gw9] [ 53%] PASSED test_any_automated_install.py::test_IPv6_only_link_local 
[gw7] [ 55%] PASSED test_any_automated_install.py::test_FTL_detect_armv8a_no_errors 
[gw10] [ 58%] PASSED test_centos_fedora_common_support.py::test_selinux_enforcing_exit 
[gw1] [ 60%] PASSED test_centos_fedora_common_support.py::test_selinux_permissive 
[gw15] [ 62%] PASSED test_centos_fedora_common_support.py::test_selinux_disabled 
[gw13] [ 65%] PASSED test_any_automated_install.py::test_os_check_passes 
[gw6] [ 67%] PASSED test_any_automated_install.py::test_FTL_detect_armv6l_no_errors 
[gw0] [ 69%] PASSED test_fedora_support.py::test_epel_and_remi_not_installed_fedora 
test_any_automated_install.py::test_FTL_detect_x86_64_no_errors 
[gw4] [ 72%] PASSED test_any_automated_install.py::test_FTL_detect_aarch64_no_errors 
test_any_automated_install.py::test_FTL_detect_armv7l_no_errors 
test_any_automated_install.py::test_package_manager_has_installer_deps 
test_any_utils.py::test_getFTLAPIPort_custom 
[gw4] [ 74%] PASSED test_any_utils.py::test_getFTLAPIPort_custom 
[gw5] [ 76%] PASSED test_any_automated_install.py::test_FTL_detect_armv5te_no_errors 
[gw13] [ 79%] PASSED test_any_automated_install.py::test_package_manager_has_installer_deps 
[gw14] [ 81%] PASSED test_any_automated_install.py::test_package_manager_has_web_deps 
[gw6] [ 83%] PASSED test_any_automated_install.py::test_FTL_detect_armv7l_no_errors 
[gw7] [ 86%] PASSED test_any_automated_install.py::test_FTL_detect_x86_64_no_errors 
[gw8] [ 88%] PASSED test_any_automated_install.py::test_FTL_download_aarch64_no_errors 
[gw12] [ 90%] PASSED test_any_automated_install.py::test_os_check_fails 
[gw2] [ 93%] PASSED test_any_automated_install.py::test_installPihole_fresh_install_readableFiles 
test_any_utils.py::test_key_removal_works 
[gw2] [ 95%] PASSED test_any_utils.py::test_key_removal_works 
[gw3] [ 97%] PASSED test_any_automated_install.py::test_installPihole_fresh_install_readableBlockpage[True] 
test_any_automated_install.py::test_update_package_cache_success_no_errors 
[gw3] [100%] PASSED test_any_automated_install.py::test_update_package_cache_success_no_errors 

============================================================================== 43 passed in 86.62s (0:01:26) ===============================================================================
.pkg: _exit /home/adam/pi-hole/test> python /home/adam/.local/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
  py3: OK (99.69=setup[8.04]+cmd[4.81,86.84] seconds)
  congratulations :) (99.72 seconds)
adam@adam-pc:~/pi-hole$ tox -c test/tox.fedora_36.ini 
.pkg: _optional_hooks /home/adam/pi-hole/test> python /home/adam/.local/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: get_requires_for_build_sdist /home/adam/pi-hole/test> python /home/adam/.local/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: prepare_metadata_for_build_wheel /home/adam/pi-hole/test> python /home/adam/.local/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: build_sdist /home/adam/pi-hole/test> python /home/adam/.local/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
py3: install_package /home/adam/pi-hole/test> python -I -m pip install --force-reinstall --no-deps /home/adam/pi-hole/test/.tox/.tmp/package/2/UNKNOWN-0.0.0.tar.gz
py3: commands[0] /home/adam/pi-hole/test> docker build -f _fedora_36.Dockerfile -t pytest_pihole:test_container ../
[+] Building 3.2s (11/11) FINISHED 

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
py3: commands[1] /home/adam/pi-hole/test> pytest -vv -n auto ./test_any_automated_install.py ./test_any_utils.py ./test_centos_fedora_common_support.py ./test_fedora_support.py
=================================================================================== test session starts ====================================================================================
platform linux -- Python 3.10.6, pytest-7.2.0, pluggy-1.0.0 -- /home/adam/pi-hole/test/.tox/py3/bin/python
cachedir: .tox/py3/.pytest_cache
rootdir: /home/adam/pi-hole/test
plugins: testinfra-7.0.0, xdist-3.1.0
[gw0] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw1] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw2] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw3] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw4] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw5] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw6] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw7] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw8] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw9] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw10] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw11] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw12] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw13] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw14] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw15] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw0] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                     
[gw1] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                        
[gw2] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                         
[gw3] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                              
[gw4] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                 
[gw5] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                  
[gw6] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                       
[gw7] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                          
[gw8] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                           
[gw9] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                
[gw10] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                  
[gw11] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                   
[gw12] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                      
[gw13] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                           
[gw14] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                              
[gw15] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                               
gw0 [43] / gw1 [43] / gw2 [43] / gw3 [43] / gw4 [43] / gw5 [43] / gw6 [43] / gw7 [43] / gw8 [43] / gw9 [43] / gw10 [43] / gw11 [43] / gw12 [43] / gw13 [43] / gw14 [43] / gw15 [43]
scheduling tests via LoadScheduling

test_any_automated_install.py::test_installPiholeWeb_fresh_install_no_errors 
test_any_automated_install.py::test_update_package_cache_failure_no_errors 
test_any_automated_install.py::test_supported_package_manager 
test_any_automated_install.py::test_setupVars_saved_to_file 
test_any_automated_install.py::test_installPihole_fresh_install_readableBlockpage[True] 
test_any_automated_install.py::test_validate_ip 
test_any_automated_install.py::test_package_manager_has_pihole_deps 
test_any_automated_install.py::test_FTL_detect_armv4t_no_errors 
test_any_automated_install.py::test_os_check_passes 
test_any_automated_install.py::test_FTL_detect_armv6l_no_errors 
test_any_utils.py::test_key_val_replacement_works 
test_any_automated_install.py::test_FTL_detect_armv8a_no_errors 
test_any_automated_install.py::test_FTL_binary_installed_and_responsive_no_errors 
test_any_automated_install.py::test_IPv6_GUA_ULA_test 
test_any_automated_install.py::test_IPv6_only_ULA 
test_any_automated_install.py::test_FTL_detect_unknown_no_errors 
[gw1] [  2%] PASSED test_any_automated_install.py::test_setupVars_saved_to_file 
[gw15] [  4%] PASSED test_any_utils.py::test_key_val_replacement_works 
[gw11] [  6%] PASSED test_any_automated_install.py::test_IPv6_GUA_ULA_test 
[gw0] [  9%] PASSED test_any_automated_install.py::test_supported_package_manager 
test_any_automated_install.py::test_selinux_not_detected 
[gw2] [ 11%] PASSED test_any_automated_install.py::test_installPiholeWeb_fresh_install_no_errors 
test_any_utils.py::test_key_addition_works 
test_any_automated_install.py::test_IPv6_ULA_GUA_test 
test_any_automated_install.py::test_setupVars_are_sourced_to_global_scope 
test_any_automated_install.py::test_installPihole_fresh_install_readableFiles 
[gw10] [ 13%] PASSED test_any_automated_install.py::test_IPv6_only_ULA 
[gw4] [ 16%] PASSED test_any_automated_install.py::test_update_package_cache_failure_no_errors 
test_any_automated_install.py::test_IPv6_only_GUA 
[gw1] [ 18%] PASSED test_any_automated_install.py::test_selinux_not_detected 
test_any_automated_install.py::test_FTL_detect_aarch64_no_errors 
[gw15] [ 20%] PASSED test_any_utils.py::test_key_addition_works 
[gw0] [ 23%] PASSED test_any_automated_install.py::test_setupVars_are_sourced_to_global_scope 
test_any_utils.py::test_key_removal_works 
[gw12] [ 25%] PASSED test_any_automated_install.py::test_validate_ip 
[gw11] [ 27%] PASSED test_any_automated_install.py::test_IPv6_ULA_GUA_test 
test_any_utils.py::test_getFTLAPIPort_default 
[gw10] [ 30%] PASSED test_any_automated_install.py::test_IPv6_only_GUA 
test_any_utils.py::test_getFTLAPIPort_malicious 
test_any_automated_install.py::test_os_check_fails 
test_any_utils.py::test_getFTLAPIPort_custom 
test_any_utils.py::test_getFTLPID_default 
[gw6] [ 32%] PASSED test_any_automated_install.py::test_FTL_detect_armv6l_no_errors 
[gw1] [ 34%] PASSED test_any_utils.py::test_key_removal_works 
test_any_automated_install.py::test_FTL_detect_armv7l_no_errors 
[gw15] [ 37%] PASSED test_any_utils.py::test_getFTLAPIPort_default 
[gw11] [ 39%] PASSED test_any_utils.py::test_getFTLAPIPort_custom 
test_centos_fedora_common_support.py::test_selinux_enforcing_exit 
test_centos_fedora_common_support.py::test_selinux_permissive 
[gw0] [ 41%] PASSED test_any_utils.py::test_getFTLAPIPort_malicious 
[gw10] [ 44%] PASSED test_any_utils.py::test_getFTLPID_default 
[gw5] [ 46%] PASSED test_any_automated_install.py::test_FTL_detect_armv4t_no_errors 
test_centos_fedora_common_support.py::test_selinux_disabled 
test_any_automated_install.py::test_FTL_detect_armv5te_no_errors 
[gw9] [ 48%] PASSED test_any_automated_install.py::test_FTL_binary_installed_and_responsive_no_errors 
test_any_automated_install.py::test_IPv6_only_link_local 
[gw14] [ 51%] PASSED test_any_automated_install.py::test_package_manager_has_pihole_deps 
[gw8] [ 53%] PASSED test_any_automated_install.py::test_FTL_detect_unknown_no_errors 
[gw15] [ 55%] PASSED test_centos_fedora_common_support.py::test_selinux_permissive 
[gw13] [ 58%] PASSED test_any_automated_install.py::test_os_check_passes 
[gw1] [ 60%] PASSED test_centos_fedora_common_support.py::test_selinux_enforcing_exit 
[gw0] [ 62%] PASSED test_centos_fedora_common_support.py::test_selinux_disabled 
[gw4] [ 65%] PASSED test_any_automated_install.py::test_FTL_detect_aarch64_no_errors 
[gw7] [ 67%] PASSED test_any_automated_install.py::test_FTL_detect_armv8a_no_errors 
test_any_automated_install.py::test_package_manager_has_web_deps 
[gw9] [ 69%] PASSED test_any_automated_install.py::test_IPv6_only_link_local 
test_any_automated_install.py::test_FTL_download_aarch64_no_errors 
test_any_automated_install.py::test_package_manager_has_installer_deps 
test_any_utils.py::test_getFTLPIDFile_and_getFTLPID_custom 
test_any_automated_install.py::test_FTL_detect_x86_64_no_errors 
[gw6] [ 72%] PASSED test_any_automated_install.py::test_FTL_detect_armv7l_no_errors 
[gw4] [ 74%] PASSED test_any_utils.py::test_getFTLPIDFile_and_getFTLPID_custom 
[gw5] [ 76%] PASSED test_any_automated_install.py::test_FTL_detect_armv5te_no_errors 
[gw12] [ 79%] PASSED test_any_automated_install.py::test_os_check_fails 
test_fedora_support.py::test_epel_and_remi_not_installed_fedora 
[gw12] [ 81%] PASSED test_fedora_support.py::test_epel_and_remi_not_installed_fedora 
[gw13] [ 83%] PASSED test_any_automated_install.py::test_package_manager_has_installer_deps 
[gw7] [ 86%] PASSED test_any_automated_install.py::test_FTL_detect_x86_64_no_errors 
[gw14] [ 88%] PASSED test_any_automated_install.py::test_package_manager_has_web_deps 
[gw8] [ 90%] PASSED test_any_automated_install.py::test_FTL_download_aarch64_no_errors 
[gw2] [ 93%] PASSED test_any_automated_install.py::test_installPihole_fresh_install_readableFiles 
test_any_utils.py::test_getFTLPIDFile_default 
[gw3] [ 95%] PASSED test_any_automated_install.py::test_installPihole_fresh_install_readableBlockpage[True] 
[gw2] [ 97%] PASSED test_any_utils.py::test_getFTLPIDFile_default 
test_any_automated_install.py::test_update_package_cache_success_no_errors 
[gw3] [100%] PASSED test_any_automated_install.py::test_update_package_cache_success_no_errors 

============================================================================== 43 passed in 82.76s (0:01:22) ===============================================================================
.pkg: _exit /home/adam/pi-hole/test> python /home/adam/.local/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
  py3: OK (87.84=setup[1.45]+cmd[3.46,82.93] seconds)
  congratulations :) (87.87 seconds)
adam@adam-pc:~/pi-hole$ tox -c test/tox.fedora_37.ini 
.pkg: _optional_hooks /home/adam/pi-hole/test> python /home/adam/.local/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: get_requires_for_build_sdist /home/adam/pi-hole/test> python /home/adam/.local/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: prepare_metadata_for_build_wheel /home/adam/pi-hole/test> python /home/adam/.local/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: build_sdist /home/adam/pi-hole/test> python /home/adam/.local/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
py3: install_package /home/adam/pi-hole/test> python -I -m pip install --force-reinstall --no-deps /home/adam/pi-hole/test/.tox/.tmp/package/3/UNKNOWN-0.0.0.tar.gz
py3: commands[0] /home/adam/pi-hole/test> docker build -f _fedora_37.Dockerfile -t pytest_pihole:test_container ../
[+] Building 57.7s (12/12) FINISHED 

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
py3: commands[1] /home/adam/pi-hole/test> pytest -vv -n auto ./test_any_automated_install.py ./test_any_utils.py ./test_centos_fedora_common_support.py ./test_fedora_support.py
=================================================================================== test session starts ====================================================================================
platform linux -- Python 3.10.6, pytest-7.2.0, pluggy-1.0.0 -- /home/adam/pi-hole/test/.tox/py3/bin/python
cachedir: .tox/py3/.pytest_cache
rootdir: /home/adam/pi-hole/test
plugins: testinfra-7.0.0, xdist-3.1.0
[gw0] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw1] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw2] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw3] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw4] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw5] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw6] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw7] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw8] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw9] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                             
[gw10] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw11] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw12] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw13] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw14] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw15] linux Python 3.10.6 cwd: /home/adam/pi-hole/test                                                                            
[gw0] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                     
[gw1] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                      
[gw2] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                           
[gw3] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                              
[gw4] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                 
[gw5] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                    
[gw6] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                       
[gw7] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                          
[gw8] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                             
[gw9] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                
[gw10] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                
[gw11] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                     
[gw12] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                      
[gw13] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                           
[gw14] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                              
[gw15] Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]                                                                                                                 
gw0 [43] / gw1 [43] / gw2 [43] / gw3 [43] / gw4 [43] / gw5 [43] / gw6 [43] / gw7 [43] / gw8 [43] / gw9 [43] / gw10 [43] / gw11 [43] / gw12 [43] / gw13 [43] / gw14 [43] / gw15 [43]
scheduling tests via LoadScheduling

test_any_automated_install.py::test_setupVars_saved_to_file 
test_any_automated_install.py::test_FTL_detect_armv8a_no_errors 
test_any_automated_install.py::test_FTL_detect_armv6l_no_errors 
test_any_automated_install.py::test_FTL_binary_installed_and_responsive_no_errors 
test_any_automated_install.py::test_supported_package_manager 
test_any_automated_install.py::test_update_package_cache_failure_no_errors 
test_any_automated_install.py::test_package_manager_has_pihole_deps 
test_any_automated_install.py::test_installPiholeWeb_fresh_install_no_errors 
test_any_automated_install.py::test_validate_ip 
test_any_automated_install.py::test_installPihole_fresh_install_readableBlockpage[True] 
test_any_automated_install.py::test_IPv6_only_ULA 
test_any_automated_install.py::test_FTL_detect_unknown_no_errors 
test_any_automated_install.py::test_FTL_detect_armv4t_no_errors 
test_any_automated_install.py::test_IPv6_GUA_ULA_test 
test_any_automated_install.py::test_os_check_passes 
test_any_utils.py::test_key_val_replacement_works 
[gw11] [  2%] PASSED test_any_automated_install.py::test_IPv6_GUA_ULA_test 
[gw15] [  4%] PASSED test_any_utils.py::test_key_val_replacement_works 
[gw2] [  6%] PASSED test_any_automated_install.py::test_installPiholeWeb_fresh_install_no_errors 
[gw4] [  9%] PASSED test_any_automated_install.py::test_update_package_cache_failure_no_errors 
[gw1] [ 11%] PASSED test_any_automated_install.py::test_setupVars_saved_to_file 
[gw0] [ 13%] PASSED test_any_automated_install.py::test_supported_package_manager 
[gw10] [ 16%] PASSED test_any_automated_install.py::test_IPv6_only_ULA 
test_any_automated_install.py::test_IPv6_ULA_GUA_test 
test_any_utils.py::test_key_addition_works 
test_any_automated_install.py::test_installPihole_fresh_install_readableFiles 
test_any_automated_install.py::test_FTL_detect_aarch64_no_errors 
test_any_automated_install.py::test_selinux_not_detected 
test_any_automated_install.py::test_setupVars_are_sourced_to_global_scope 
test_any_automated_install.py::test_IPv6_only_GUA 
[gw11] [ 18%] PASSED test_any_automated_install.py::test_IPv6_ULA_GUA_test 
test_any_utils.py::test_key_removal_works 
[gw10] [ 20%] PASSED test_any_automated_install.py::test_IPv6_only_GUA 
[gw1] [ 23%] PASSED test_any_automated_install.py::test_selinux_not_detected 
[gw15] [ 25%] PASSED test_any_utils.py::test_key_addition_works 
[gw9] [ 27%] PASSED test_any_automated_install.py::test_FTL_binary_installed_and_responsive_no_errors 
test_any_utils.py::test_getFTLPIDFile_and_getFTLPID_custom 
test_any_utils.py::test_getFTLPIDFile_default 
[gw14] [ 30%] PASSED test_any_automated_install.py::test_package_manager_has_pihole_deps 
test_any_utils.py::test_getFTLAPIPort_default 
test_any_automated_install.py::test_IPv6_only_link_local 
[gw0] [ 32%] PASSED test_any_automated_install.py::test_setupVars_are_sourced_to_global_scope 
[gw12] [ 34%] PASSED test_any_automated_install.py::test_validate_ip 
test_any_automated_install.py::test_package_manager_has_web_deps 
[gw11] [ 37%] PASSED test_any_utils.py::test_key_removal_works 
[gw5] [ 39%] PASSED test_any_automated_install.py::test_FTL_detect_armv4t_no_errors 
[gw7] [ 41%] PASSED test_any_automated_install.py::test_FTL_detect_armv8a_no_errors 
[gw10] [ 44%] PASSED test_any_utils.py::test_getFTLPIDFile_and_getFTLPID_custom 
test_any_automated_install.py::test_os_check_fails 
test_any_utils.py::test_getFTLPID_default 
[gw8] [ 46%] PASSED test_any_automated_install.py::test_FTL_detect_unknown_no_errors 
[gw6] [ 48%] PASSED test_any_automated_install.py::test_FTL_detect_armv6l_no_errors 
test_any_automated_install.py::test_FTL_detect_armv5te_no_errors 
[gw15] [ 51%] PASSED test_any_utils.py::test_getFTLAPIPort_default 
[gw1] [ 53%] PASSED test_any_utils.py::test_getFTLPIDFile_default 
test_centos_fedora_common_support.py::test_selinux_enforcing_exit 
test_any_automated_install.py::test_FTL_detect_x86_64_no_errors 
test_any_automated_install.py::test_FTL_download_aarch64_no_errors 
test_any_automated_install.py::test_FTL_detect_armv7l_no_errors 
[gw9] [ 55%] PASSED test_any_automated_install.py::test_IPv6_only_link_local 
test_centos_fedora_common_support.py::test_selinux_permissive 
[gw4] [ 58%] PASSED test_any_automated_install.py::test_FTL_detect_aarch64_no_errors 
test_fedora_support.py::test_epel_and_remi_not_installed_fedora 
test_centos_fedora_common_support.py::test_selinux_disabled 
test_any_utils.py::test_getFTLAPIPort_malicious 
[gw13] [ 60%] PASSED test_any_automated_install.py::test_os_check_passes 
test_any_automated_install.py::test_package_manager_has_installer_deps 
[gw0] [ 62%] PASSED test_any_utils.py::test_getFTLPID_default 
[gw11] [ 65%] PASSED test_centos_fedora_common_support.py::test_selinux_enforcing_exit 
[gw10] [ 67%] PASSED test_centos_fedora_common_support.py::test_selinux_permissive 
[gw1] [ 69%] PASSED test_centos_fedora_common_support.py::test_selinux_disabled 
[gw15] [ 72%] PASSED test_fedora_support.py::test_epel_and_remi_not_installed_fedora 
[gw4] [ 74%] PASSED test_any_utils.py::test_getFTLAPIPort_malicious 
[gw5] [ 76%] PASSED test_any_automated_install.py::test_FTL_detect_armv5te_no_errors 
[gw13] [ 79%] PASSED test_any_automated_install.py::test_package_manager_has_installer_deps 
[gw12] [ 81%] PASSED test_any_automated_install.py::test_os_check_fails 
[gw14] [ 83%] PASSED test_any_automated_install.py::test_package_manager_has_web_deps 
[gw7] [ 86%] PASSED test_any_automated_install.py::test_FTL_detect_x86_64_no_errors 
[gw6] [ 88%] PASSED test_any_automated_install.py::test_FTL_detect_armv7l_no_errors 
[gw8] [ 90%] PASSED test_any_automated_install.py::test_FTL_download_aarch64_no_errors 
[gw3] [ 93%] PASSED test_any_automated_install.py::test_installPihole_fresh_install_readableBlockpage[True] 
test_any_automated_install.py::test_update_package_cache_success_no_errors 
[gw3] [ 95%] PASSED test_any_automated_install.py::test_update_package_cache_success_no_errors 
[gw2] [ 97%] PASSED test_any_automated_install.py::test_installPihole_fresh_install_readableFiles 
test_any_utils.py::test_getFTLAPIPort_custom 
[gw2] [100%] PASSED test_any_utils.py::test_getFTLAPIPort_custom 

============================================================================== 43 passed in 89.45s (0:01:29) ===============================================================================
.pkg: _exit /home/adam/pi-hole/test> python /home/adam/.local/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
  py3: OK (149.06=setup[1.43]+cmd[58.00,89.63] seconds)
  congratulations :) (149.09 seconds)
```

Reply:
OK.  I removed that standalone commit.